drivers/storage/portworx/component/windows.go

```diff
@@ -0,0 +1,163 @@
+package component
+
+import (
+	"context"
+	commonerrors "errors"
+	version "github.com/hashicorp/go-version"
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	"github.com/libopenstorage/operator/pkg/util"
+	"github.com/libopenstorage/operator/pkg/util/k8s"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+const (
+	WindowsComponentName     = "Windows Node"
+	WindowsDaemonSetName     = "csi-pwx-node-win"
+	WindowsStorageClass      = "px-csi-win"
+	WindowsDaemonSetFileName = "win.yaml"
+)
+
+type windows struct {
+	client        client.Client
+	isCreated     bool
+	isWindowsNode bool
+}
+
+func (w *windows) Initialize(
+	k8sClient client.Client,
+	_ version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	w.client = k8sClient
+}
+
+func (w *windows) Name() string {
+	return WindowsComponentName
+}
+
+func (w *windows) Priority() int32 {
+	return DefaultComponentPriority
+}
+
+func (w *windows) IsPausedForMigration(cluster *corev1.StorageCluster) bool {
+	return util.ComponentsPausedForMigration(cluster)
+}
+
+func (w *windows) IsEnabled(cluster *corev1.StorageCluster) bool {
+	w.isWindowsNode = true
```

Comment:
Isn't there going to be a way to automatically check if this is windows or not? We shouldn't be running this DS on non-windows envs, right?

Reply:
yes added the check